include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/AutoPreviewScreenshotsExtension.kt

```diff
@@ -0,0 +1,170 @@
+  )
+  project.dependencies.add(
+    configurationName,
+    "io.github.takahirom.roborazzi:roborazzi:$roborazziVersion"
+  )
+  project.dependencies.add(
+    configurationName,
+    "junit:junit:${BuildConfig.libraryVersionsMap["junit"]}"
+  )
+  project.dependencies.add(
+    configurationName,
+    "org.robolectric:robolectric:${BuildConfig.libraryVersionsMap["robolectric"]}"
+  )
+
+  // For ComposablePreviewScanner
+  project.repositories.add(project.repositories.maven { it.setUrl("https://jitpack.io") })
+  project.repositories.add(project.repositories.mavenCentral())
+  project.repositories.add(project.repositories.google())
+  project.dependencies.add(
+    configurationName,
+    "com.github.sergio-sastre.ComposablePreviewScanner:android:${BuildConfig.libraryVersionsMap["composable-preview-scanner"]}"
+  )
+}
+
+abstract class GeneratePreviewScreenshotTestsTask : DefaultTask() {
+  @get:OutputDirectory
+  abstract val outputDir: DirectoryProperty
+
+  @get:Input
+  var scanPackageTrees: ListProperty<String> = project.objects.listProperty(String::class.java)
+
+  @TaskAction
+  fun generateTests() {
+    val testDir = outputDir.get().asFile
+    testDir.mkdirs()
+
+    val packagesExpr = scanPackageTrees.get().joinToString(", ") { "\"$it\"" }
+    val scanPackageTreeExpr = ".scanPackageTrees($packagesExpr)"
+
+    File(testDir, "PreviewParameterizedTests.kt").writeText(
+      """
+            import org.junit.Test
+            import org.junit.runner.RunWith
+            import org.robolectric.ParameterizedRobolectricTestRunner
+            import org.robolectric.annotation.Config
+            import org.robolectric.annotation.GraphicsMode
+            import com.github.takahirom.roborazzi.captureRoboImage
+            import sergio.sastre.composable.preview.scanner.core.preview.ComposablePreview
+            import sergio.sastre.composable.preview.scanner.android.AndroidPreviewInfo
+            import sergio.sastre.composable.preview.scanner.android.AndroidComposablePreviewScanner
+            import com.github.takahirom.roborazzi.DEFAULT_ROBORAZZI_OUTPUT_DIR_PATH
+
+
+            @RunWith(ParameterizedRobolectricTestRunner::class)
+            @GraphicsMode(GraphicsMode.Mode.NATIVE)
+            class PreviewParameterizedTests(
+                private val preview: ComposablePreview<AndroidPreviewInfo>,
+            ) {
+
+                companion object {
+                    @JvmStatic
+                    @ParameterizedRobolectricTestRunner.Parameters
+                    fun values(): List<ComposablePreview<AndroidPreviewInfo>> =
+                        AndroidComposablePreviewScanner()
+                            $scanPackageTreeExpr
+                            .getPreviews()
+                }
+                
+                @GraphicsMode(GraphicsMode.Mode.NATIVE)
+                @Config(sdk = [30])
+                @Test
+                fun snapshot() {
+                    val filePath = preview.methodName + ".png"
```

Comment:
right now, the `previewInfo` parameters like fontScale, locale, etc. are not set.

I've showcased [how to set them here](https://github.com/sergio-sastre/Android-screenshot-testing-playground/blob/master/lazycolumnscreen-previews/roborazzi/src/test/java/snapshot/testing/lazycolumn_previews/roborazzi/utils/RobolectricPreviewInfosApplier.kt) which is [applied in this test](https://github.com/sergio-sastre/Android-screenshot-testing-playground/blob/master/lazycolumnscreen-previews/roborazzi/src/test/java/snapshot/testing/lazycolumn_previews/roborazzi/RoborazziComposePreviewTests.kt) :)

Still some important missing, like "device", and others like "wallpapers" are not currently possible in Robolectric as far as I know... likely worth documenting